tanner/api/api.py

```diff
@@ -34,25 +56,107 @@ async def return_snare_stats(self, snare_uuid):
         result['total_sessions'] = len(sessions)
         for sess in sessions:
             result['total_duration'] += sess['end_time'] - sess['start_time']
-            for attack in sess['attack_types']:
-                if attack in result['attack_frequency']:
-                    result['attack_frequency'][attack] += 1
+            for path in sess['paths']:
+                if path['attack_type'] in result['attack_frequency']:
+                    result['attack_frequency'][path['attack_type']] += 1
 
         return result
 
-    async def return_snare_info(self, uuid, count=-1):
+    async def return_snare_info(self, uuid):
+        """Returns JSON data that contains information about
+         all the sessions a single snare instance have.
+
+        Returns:
+            uuid [string] - Snare UUID
+        """
         query_res = []
-        try:
-            query_res = await self.redis_client.zrevrangebyscore(
-                uuid, offset=0, count=count, encoding='utf-8'
-            )
-        except aioredis.ProtocolError as connection_error:
-            self.logger.exception('Can not connect to redis %s', connection_error)
-        else:
-            if not query_res:
-                return 'Invalid SNARE UUID'
-            for (i, val) in enumerate(query_res):
-                query_res[i] = json.loads(val)
+        async with self.pg_client.acquire() as conn:
+            async with conn.cursor() as cur:
+                await cur.execute(
+                    """
+                    SELECT * FROM sessions
+                    WHERE
+                        sessions.sensor_id = '%s'
+                    """ % (uuid)
+                )
+                session = await cur.fetchall()
+                for r in session:
+                    sess = {
+                        'sess_uuid': str(r[0]),
+                        'snare_uuid': str(r[1]),
+                        'ip': r[2],
+                        'port': r[3],
+                        'location': {
+                            'country': r[4],
+                            'country_code': r[5],
+                            'city': r[6],
+                            'zip_code': r[7],
+                        },
+                        'user_agent': r[8],
+                        'start_time': r[9].timestamp(),
+                        'end_time': r[10].timestamp(),
+                        'request_per_second': r[11],
+                        'approx_time_between_requests': r[12],
+                        'accepted_paths': r[13],
+                        'errors': r[14],
+                        'hidden_links': r[15],
+                        'referrer': r[16]
+                    }
+
+                    await cur.execute(
+                        """
+                        SELECT * FROM cookies WHERE cookies.session_id = '%s'
+                        """ % (str(r[0]))
```

Comment:
What if the uuid passed to the function is wrong?  and you will not have session_id

Reply:
Isn't this depends on the insertion part? I mean the insert section have it as UUID type and if there is not seesion_id then there won't be any other data(meaning insertion wouldn't take place)

Reply:
Oh and for the passed value I'll add the check that it has to be uuid.UUID type